src/node_errors.cc

```diff
@@ -1044,6 +1050,304 @@ void TriggerUncaughtException(Isolate* isolate, const v8::TryCatch& try_catch) {
+  ASSIGN_OR_RETURN_UNWRAP(&ex, args.Holder());
+  args.GetReturnValue().Set(ex->name);
+}
+
+void DOMException::GetMessage(const FunctionCallbackInfo<Value>& args) {
+  DOMException* ex;
+  ASSIGN_OR_RETURN_UNWRAP(&ex, args.Holder());
+  args.GetReturnValue().Set(ex->message);
+}
+
+void DOMException::GetStack(const FunctionCallbackInfo<Value>& args) {
+  DOMException* ex;
+  ASSIGN_OR_RETURN_UNWRAP(&ex, args.Holder());
+  args.GetReturnValue().Set(ex->stack);
+}
+
+#define DOMEXCEPTION_CODES(V)                                                  \
+  V(IndexSize, 1)                                                              \
+  V(DOMStringSize, 2)                                                          \
+  V(HierarchyRequest, 3)                                                       \
+  V(WrongDocument, 4)                                                          \
+  V(InvalidCharacter, 5)                                                       \
+  V(NoDataAllowed, 6)                                                          \
+  V(NoModificationAllowed, 7)                                                  \
+  V(NotFound, 8)                                                               \
+  V(NotSupported, 9)                                                           \
+  V(InUseAttribute, 10)                                                        \
+  V(InvalidState, 11)                                                          \
+  V(Syntax, 12)                                                                \
+  V(InvalidModification, 13)                                                   \
+  V(Namespace, 14)                                                             \
+  V(InvalidAccess, 15)                                                         \
+  V(Validation, 16)                                                            \
+  V(TypeMismatch, 17)                                                          \
+  V(Security, 18)                                                              \
+  V(Network, 19)                                                               \
+  V(Abort, 20)                                                                 \
+  V(URLMismatch, 21)                                                           \
+  V(QuotaExceeded, 22)                                                         \
+  V(Timeout, 23)                                                               \
+  V(InvalidNodeType, 24)                                                       \
+  V(DataClone, 25)
+
+void DOMException::GetCode(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  DOMException* ex;
+  ASSIGN_OR_RETURN_UNWRAP(&ex, args.Holder());
+
+  Local<Value> name = ex->name.Get(env->isolate());
+  Utf8Value val(env->isolate(), name);
+  int ret = 0;
+
+#define V(name, code) if (strcmp(#name "Error", *val) == 0) ret = code;
+  DOMEXCEPTION_CODES(V)
+  args.GetReturnValue().Set(ret);
+#undef V
```

Comment:
```suggestion
#undef V
  args.GetReturnValue().Set(ret);
```
Setting the return value does not need to be done in the scope of V, does it?